docs/proposals/config_data.md

```diff
@@ -0,0 +1,327 @@
+# Generic Configuration Object
+
+## Abstract
+
+This proposal proposes a generic configuration object named ``ConfigData`` stored inside ``etcd`` that
+stores data used for the configuration of ``Kubernetes`` components.
```

Comment:
Actually, let's not mention Kubernetes components more than once in this doc, as just one example.